source/common/network/listen_socket_impl.h

```diff
@@ -51,9 +52,14 @@ template <typename T> class NetworkListenSocket : public ListenSocketImpl {
       : ListenSocketImpl(bind_to_port ? Network::ioHandleForAddr(T::type, address, creation_options)
                                       : nullptr,
                          address) {
+    // Allow null io handles for unit tests. Disallow closed io handles as they
+    // indicate socket creation failure, which is not yet supported for listeners.
+    if (Runtime::runtimeFeatureEnabled(
+            "envoy.restart_features.allow_client_socket_creation_failure")) {
+      RELEASE_ASSERT(!io_handle_ || io_handle_->isOpen(), "");
```

Comment:
nit: Should we put any context in the assert string?

Reply:
oh I hadn't because there was an ENVOY BUG which generally failed first but in opt mode it's 100% worth the pointer, done.

Reply:
Hm. I'm not seeing a change here?